### PR TITLE
Simply Replication Status proto conversions

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

Since v14 has happened, we can simplify the proto to and from conversions of `ReplicationStatus`. The test for the same has also been changed to only expect the IOState and SqlState to be specified.

The proto fields of io_thread_running and sql_thread_running have been marked reserved.

## Related Issue(s)

  - Follow-up to: https://github.com/vitessio/vitess/pull/10167

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by dbussink — APPROVED** [review]

**Review by mattlord — COMMENTED** [review]
Thank you for doing this! Since I think we're now essentially reverting this: https://github.com/vitessio/vitess/pull/10167

There's some other stuff we could also remove (e.g. the interim test that was added there)

File: proto/replicationdata.proto [hunk 1]
```diff
@@ -25,11 +25,9 @@ package replicationdata;
 // flavor-specific command and parsed into a Position and fields.
 message Status {
   string position = 1;
-  // These fields should be removed in Vitess 15+ and fully replaced by the io_state and sql_state fields
-  // reserved 2, 3;
-  // reserved "io_thread_running", "sql_thread_running";
```
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply proto/replicationdata.proto:30]: Is re-using a name not correct in proto?

[the remark this answers is not in the document]
Comment (deepthi) [reply proto/replicationdata.proto:30]: Names don't really matter for the wire protocol, it is the numbers. The only time names come into play is when a proto is converted to JSON. I'm iffy about whether it is a good idea to re-use names, but we have not reserved them for any of the ones we removed in the past. In that sense this PR is fine.

**Review by deepthi — APPROVED** [review]

**Comment by deepthi** [comment]
> There's some other stuff we could also remove (e.g. the interim test that was added there)

Should probably wait until this is addressed before merging.

**Review by mattlord — APPROVED** [review]
Thanks again, @GuptaManan100 !

**Pull request merged**